src/modules/auth/components/MailConfirmationTemplate/MailConfirmationTemplate.tsx

```diff
@@ -0,0 +1,126 @@
+import { useEffect, useState } from 'react';
+import { useDebounce } from 'react-use';
+
+import { isAfter } from 'date-fns';
+
+import TranslatableComponent from '../../../shared/components/TranslatableComponent';
+import { PixwayAppRoutes } from '../../../shared/enums/PixwayAppRoutes';
+import { useCompanyConfig } from '../../../shared/hooks/useCompanyConfig';
+import useRouter from '../../../shared/hooks/useRouter';
+import useTranslation from '../../../shared/hooks/useTranslation';
+import { useVerifySignUp } from '../../hooks/useVerifySignUp';
+import { AuthButton } from '../AuthButton';
+import { AuthFooter } from '../AuthFooter';
+import { AuthLayoutBase } from '../AuthLayoutBase';
+import { VerifySignUpMailSent } from '../VerifySignUpMailSent';
+import { VerifySignUpTokenExpired } from '../VerifySignUpTokenExpired';
+
+enum Steps {
+  LOADING = 1,
+  EMAIL_VERIFIED,
+  TOKEN_EXPIRED,
+  EMAIL_SENT,
+}
+
+const TIME_TO_REDIRECT_TO_HOME = 6000;
+
+const _MailConfirmationTemplate = () => {
+  const [translate] = useTranslation();
+  const { logoUrl } = useCompanyConfig();
+  const { mutate, isLoading, isSuccess, isError } = useVerifySignUp();
+  const router = useRouter();
+  const { email, token } = router.query;
+  const [step, setStep] = useState(Steps.LOADING);
+
+  const [_, cancel] = useDebounce(
+    () => {
+      if (step === Steps.EMAIL_VERIFIED) {
+        router.push(PixwayAppRoutes.SIGN_IN);
+      }
+    },
+    TIME_TO_REDIRECT_TO_HOME,
+    [step]
+  );
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  useEffect(() => cancel, []);
```

Comment:
Não vale colocar esse junto com o de baixo tendo em vista que os dois são effects de inicialização?

Reply:
Esse não é bem de inicialização.
Como ele retorna uma função, ela é executada no ciclo de clean up do useEffect
Nesse caso quando o componente for desmontado

Mas alterei pra ficar incluso no outro